libs/coin-modules/coin-bitcoin/src/synchronisation.test.ts

```diff
@@ -0,0 +1,232 @@
+    chainCode: "",
+  },
+  {
+    publicKey:
+      "xpub6BuPWhjLqutPY3pENUxwkG3NAQfjLxoHMTAkwzH13rZSzkymZrbKui1MN3KgwiNFsH4AyLM3GkodNo4bTxQXE1kvZiruVUtJLe1bMW1MBBJ",
+    bitcoinAddress: "1JepjRPgzuV4hyBTZxftBYqbAaXMygCMRV",
+    chainCode: "",
+  },
+];
+const mockSignerContext = <T>(
+  _deviceId: string,
+  _crypto: CryptoCurrency,
+  fn: (signer: BitcoinSigner) => Promise<T>,
+): Promise<T> =>
+  fn({
+    getWalletXpub: (_arg: { path: string; xpubVersion: number }): Promise<BitcoinXPub> =>
+      Promise.resolve(xpub.shift() ?? ""),
+    getWalletPublicKey: (
+      _path: string,
+      _opts?: {
+        verify?: boolean;
+        format?: AddressFormat;
+      },
+    ): Promise<BitcoinAddress> => {
+      return Promise.resolve(
+        pubKey.shift() ?? {
+          publicKey: "",
+          bitcoinAddress: "",
+          chainCode: "",
+        },
+      );
+    },
+    signMessage: (_path: string, _messageHex: string): Promise<BitcoinSignature> =>
+      Promise.resolve({
+        v: 0,
+        r: "123",
+        s: "456",
+      }),
+    splitTransaction: (
+      _transactionHex: string,
+      _isSegwitSupported: boolean | null | undefined,
+      _hasExtraData: boolean | null | undefined,
+      _additionals: Array<string> | null | undefined,
+    ): SignerTransaction => ({
+      version: Buffer.from(""),
+      inputs: [
+        {
+          prevout: Buffer.from(""),
+          script: Buffer.from(""),
+          sequence: Buffer.from(""),
+        },
+      ],
+    }),
+    createPaymentTransaction: (_arg: CreateTransaction): Promise<string> => Promise.resolve(""),
+  });
+const mockCryptoCurrency = {
+  type: "CryptoCurrency",
+  id: "bitcoin",
+  name: "Bitcoin",
+  family: "bitcoin",
+  coinType: 0,
+  scheme: "native_segwit",
+  units: [
+    {
+      name: "BTC",
+      code: "BTC",
+      magnitude: 8,
+    },
+  ],
+  explorerViews: [
+    {
+      tx: "https://blockchain.info/tx/{txid}",
+      address: "https://blockchain.info/address/{address}",
+    },
+  ],
+  ticker: "BTC",
+  explorerId: "btc",
+  color: "#FF0000",
+  managerAppName: "Bitcoin",
+} as CryptoCurrency;
+
+const mockStartSpan = jest.fn().mockReturnValue({
+  finish: jest.fn(),
+});
+describe("synchronisation", () => {
+  it("should return a function", () => {
+    const result = makeGetAccountShape(mockSignerContext, mockStartSpan);
+    expect(typeof result).toBe("function");
+  });
+
+  it("should return an account shape with the correct properties", async () => {
+    const getAccountShape = makeGetAccountShape(mockSignerContext, mockStartSpan);
+    // getAccountCurrency.mockReturnValue({ ticker: "BTC" });
+    // getAccountUnit.mockReturnValue({ magnitude: 8 });
+    const mockAccount = createFixtureAccount();
+    mockAccount.id =
+      "js:2:bitcoin:xpub6DM4oxVnZiePFvQMu1RJLQwWUzZQP3UNaLqrGcbJQkAJZYdiRoRivHULWoYN3zBYU4mJRpM3WrGaqo1kS8Q2XFfd9E3QEc9P3MKHwbHz9LB:native_segwit";
+    const result = await getAccountShape(
```

Comment:
some questions: 
you've written the "getAccountShape" test for one account. Why do you need so many xpub in this file?
Do you mock the explorer responses for this account sync or you perform the test with Ledger explorer here?

Reply:
We don't need that many xpubs, I've removed most of those and only use 1 as a fixture for the tests